src/p2p_networking.rs

```diff
@@ -1,25 +1,50 @@
-use crate::errors::RuntimeError;
+use crate::async_runtime::Handle;
+use crate::errors::{LipaError, MapToLipaError, RuntimeError};
 use crate::{NodeAddress, PeerManager};
 use bitcoin::secp256k1::PublicKey;
-use log::debug;
+use log::{debug, error, trace};
+use std::fmt;
 use std::net::SocketAddr;
 use std::str::FromStr;
 use std::sync::Arc;
 use std::task::Poll;
 use std::time::Duration;
+use tokio::task::JoinHandle;
 use tokio::time::sleep;
 
 pub(crate) struct P2pConnections {}
 
 impl P2pConnections {
-    pub async fn connect_peer(
+    pub fn run_bg_connector(
         peer: &NodeAddress,
+        handle1: Handle,
+        peer_mgr: &Arc<PeerManager>,
```

Comment:
To maintain consistency and readability, I would prefer we use `peer_manager`

Reply:
Oh, I see that maybe you used this variation to distinguish from the inner `peer_manager`? I would still avoid doing so because we can use shadowing.

Reply:
Fixed: 440c22abbf62ab577edd2f83b871a1af427ca4ef